src/main/java/data/Shot.java

```diff
@@ -97,18 +103,43 @@ public int compareTo(Shot other) {
      * @return true when shots are overlapping, false when there are not overlapping
      */
     public boolean areOverlapping(Shot other, double movementOffset) {
+        boolean result = false;
+
+        // Other shot starts during this shot
         if (other.getBeginCount() > getBeginCount() - movementOffset
-                && other.getBeginCount() < getEndCount()) {
-            return true;
+                && other.getBeginCount() - movementOffset < getEndCount()) {
+            result = true;
         }
+
+        // This shot starts during other shot
         if (other.getEndCount() > getBeginCount() - movementOffset
                 && other.getEndCount() < getEndCount()) {
-            return true;
+            result = true;
         }
+
+        // This shot entirely in other shot or the other way around
         if (other.getBeginCount() <= getBeginCount() && other.getEndCount() >= getEndCount()
             || getBeginCount() < other.getBeginCount() && getEndCount() > other.getEndCount()) {
-            return true;
+            result = true;
+        }
+
+        if (result) {
+            // Add to collideswith if it is not in there
+            if (!this.collidesWith.contains(other)) {
+                this.collidesWith.add(other);
+            }
+            if (!other.getCollidesWith().contains(this)) {
+                other.getCollidesWith().add(this);
+            }
+        } else {
+            // Remove from collideswith if it is in there, doesn't collide anymore
+            if (this.collidesWith.contains(other)) {
+                this.collidesWith.remove(other);
+            }
+            if (other.getCollidesWith().contains(this)) {
+                other.getCollidesWith().remove(this);
+            }
```

Comment:
Since it is not used, I think it could be removed. I added it previously to have a way to avoid a lot of overhead during inefficient checks and methods.